tgui/packages/tgui/interfaces/Mecha/MechStatPane.tsx

```diff
@@ -0,0 +1,185 @@
+import { useBackend } from '../../backend';
+import { Stack, Button, Section, Box, ProgressBar, LabeledList } from '../../components';
+import { KelvinZeroCelcius, OperatorData } from './data';
+import { toFixed } from 'common/math';
+
+export const MechStatPane = (props, context) => {
+  const { act, data } = useBackend<OperatorData>(context);
+  const {
+    name,
+    integrity,
+    air_source,
+    cabin_pressure,
+    cabin_dangerous_highpressure,
+    cabin_temp,
+    mecha_flags,
+    mechflag_keys,
+    port_connected,
+  } = data;
+  return (
+    <Stack fill vertical>
+      <Stack.Item>
+        <Section
+          title={name}
+          buttons={(
+            <Button onClick={() => act('changename')}>
+              Rename
+            </Button>
+          )} />
+      </Stack.Item>
+      <Stack.Item>
+        <Section title="Status">
+          <LabeledList>
+            <LabeledList.Item label="Integrity">
+              <ProgressBar
+                ranges={{
+                  good: [0.5, Infinity],
+                  average: [0.25, 0.5],
+                  bad: [-Infinity, 0.25],
+                }}
+                value={integrity} />
+            </LabeledList.Item>
+            <LabeledList.Item label="Power">
+              <PowerBar />
+            </LabeledList.Item>
+          </LabeledList>
+        </Section>
+      </Stack.Item>
+      <Stack.Item>
+        <Section title="Enviromental Data">
+          <LabeledList>
+            <LabeledList.Item label="Air Source">
+              <Button onClick={() => act('toggle_airsource')}>
+                {air_source}
+              </Button>
+            </LabeledList.Item>
+            <LabeledList.Item label="Cabin pressure">
+              <Box color={cabin_pressure>cabin_dangerous_highpressure?"red":null}>
+                {cabin_pressure} kPa
+              </Box>
+            </LabeledList.Item>
+            <LabeledList.Item label="Cabin temperature">
+              <Box>
+                {GetTempFormat(cabin_temp)}
```

Comment:
you might need to double check with @stylemistake that it's ok to use a function like this inside the template.

Reply:
Surely `GetTempFormat` is a pure function? Pure functions in render are fine.